src/group_impl.h

```diff
@@ -241,6 +241,18 @@ static int secp256k1_ge_set_xo_var(secp256k1_ge *r, const secp256k1_fe *x, int o
 
 }
 
+static void secp256k1_ge_even_y(secp256k1_ge *r, int *y_parity) {
+    if (y_parity != NULL) {
+        *y_parity = 0;
+    }
+    if (secp256k1_fe_is_odd(&r->y)) {
+        secp256k1_ge_neg(r, r);
```

Comment:
I don't know if you care, but technically it is faster to call `secp256k1_fe_negate(&r->y, &r->y, 1)` here.  `fe_is_odd` already requires `r->y` to be fully normalized as a prerequisite.

Reply:
I don't care about the 4 nanoseconds, but I care about removing unnecessary stuff. Better to use fe_negate. Also, this function seems to be out of place in group.h, because no other function there assumes that the coordinates are normalized. Perhaps best to move this function into extrakeys/main_impl.h